File: Proxy Pattern/1. Minimal/Python/proxy_subject.py
from abstract_subject import AbstractSubject
from real_subject import RealSubject

class ProxySubject(AbstractSubject):
    def __init__(self):
        self.real_subject = None

    def doSomething(self):
        if self.real_subject is None:
            self.real_subject = RealSubject()

        self.real_subject.doSomething()

File: Proxy Pattern/1. Minimal/Java/ProxySubject.java
public class ProxySubject implements AbstractSubject {
    private RealSubject realSubject;

    @Override
    public void doSomething() {
        if (this.realSubject == null) {
            this.realSubject = new RealSubject();
        }

        this.realSubject.doSomething();
    }
}
